src/wallet/signer.ts

```diff
@@ -0,0 +1,222 @@
+    verifyBaseTransaction(tx as unknown as Record<string, unknown>);
+
+    if (tx.SigningPubKey !== "") {
+      throw new ValidationError(
+        "For multisigning the transaction must include the SigningPubKey field as an empty string."
+      );
+    }
+
+    if (tx.Signers == null) {
+      throw new ValidationError(
+        "For multisigning the transaction must include a Signers field containing an array of signatures."
+      );
+    }
+  });
+
+  const encodedTransactions: string[] = transactions.map((txOrBlob) =>
+    getEncodedTransaction(txOrBlob)
+  );
+
+  return combine(encodedTransactions);
+}
+
+/**
+ * Creates a signature that can be used to redeem a specific amount of XRP from a payment channel.
+ *
+ * @param wallet - The account that will sign for this payment channel.
+ * @param channelId - An id for the payment channel to redeem XRP from.
+ * @param amount - The amount in drops to redeem.
+ * @returns A signature that can be used to redeem a specific amount of XRP from a payment channel.
+ */
+function authorizeChannel(
+  wallet: Wallet,
+  channelId: string,
+  amount: string
+): string {
+  const signingData = encodeForSigningClaim({
+    channel: channelId,
+    amount,
+  });
+
+  return signWithKeypair(signingData, wallet.privateKey);
+}
+
+/**
+ * Verifies that the given transaction has a valid signature based on public-key encryption.
+ *
+ * @param tx - A transaction to verify the signature of. (Can be in object or encoded string format).
+ * @returns Returns true if tx has a valid signature, and returns false otherwise.
+ */
+function verify(tx: Transaction | string): boolean {
+  const decodedTx: Transaction = getDecodedTransaction(tx);
+  return verifySignature(
+    encodeForSigning(decodedTx),
+    decodedTx.TxnSignature,
+    decodedTx.SigningPubKey
+  );
+}
+
+/**
+ * Takes a collection of signedTransactions with the same underlying transaction and produces a
+ * Transaction with all Signers. It then signs that Transaction and gives an id based on the combined Transaction.
+ *
+ * @param signedTransactions - A collection of the same transaction signed by different signers. The only difference
+ * between the elements of signedTransactions should be the Signers field.
+ * @returns An object with the combined transaction (now having a sorted list of all signers) which is encoded, along
+ * with a transaction id based on the combined transaction.
+ */
+function combine(signedTransactions: string[]): SignedTransaction {
+  const transactions: Transaction[] = signedTransactions.map((tx: string) => {
+    return getDecodedTransaction(tx);
+  });
+
+  // This will throw a more clear error for JS users if there's a problem with any of the transaction's formatting
+  transactions.forEach((tx) =>
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Casting here let's us do type validation for JS
+    verifyBaseTransaction(tx as unknown as Record<string, unknown>)
```

Comment:
Why do we verify all of the transactions in 2 separate places? We should only need to do it once. (see comment above about combining the two methods into one)